### DISCUSSION TITLE
meaning of gene % prerank output

### DISCUSSION THREAD

**Discussion by idriskb** [zqfang/GSEApy#210] [Q&A] [answered]
Hello,

I am running prerank enrichment on my own ranking method. I am struggling to understand what does gene% mean in the output dataframe. I read in the documentation that:
tag %: Percent of gene set before running enrichment peak (ES),
gene %: Percent of gene list before running enrichment peak (ES),

But still I cannot understand it

**Comment by zqfang** [comment, marked as answer]
so, the 

$$ \text{gene \\%} = \text{(the position of the gene coresponding to enrichment score peak)} \div \text{(number of all genes in ranking list)}$$

and 


$$ \text{tag \\%}  = \text{( number of leading genes)} \div \text{(number of genes in certain pathway that overalpped with the input ranking list)}$$

@Cher-HAN , Thanks for the correction (This metric is same to the GSEA_R)

**Reply by idriskb** [reply]
Thank you for the answer.

Just a small detail, what do you mean exactly by "leading genes" ?

**Reply by kirillkrs** [reply]
This is list of genes before maximum ES value
This is how I understood this

**Reply by zqfang** [reply]
Yes. Exactly @kirillkrs

**Reply by Cher-HAN** [reply]
Hi, 
I am also trying to understand the meaning of Gene% and Tag%. Unluckliy, I am still confused after reading the answer because the formulas above not fit my GSEA results. I attched my example results below used to explain.
As I understand:
number of leading genes = 18
number of all genes in ranking list = 389
number of genes in this GO term that overlap my gene list = 20.
Then, Gene % = 18/389 which is 4.6%. But the results from gseapy_prerank shows the Gene % is 51.16%

I am curious about what is wrong in my understanding/calculation?

Kind regards,
Cher

![81697816973_ pic](https://github.com/zqfang/GSEApy/assets/61700436/73cf34cd-d6d5-45bc-b338-f1f5734ef7d0)

**Reply by zqfang** [reply]
looks like something is funny here. so how's your df_rnk look like? @Cher-HAN

**Reply by Cher-HAN** [reply]
Hi, zq. 
The score used to rank in df_rnk is from 0 - 1. I draw a distribution boxplot and attched below. 
![5751697837658_ pic](https://github.com/zqfang/GSEApy/assets/61700436/e97bb2b0-d111-46b7-9790-bd5c03f80d50)

**Reply by zqfang** [reply]
I don't have a clue what's going on here. Do you have duplicated gene names ? Can you set verbose=True or share the logging file ?

**Reply by Cher-HAN** [reply]
Thanks for the reply. 
I am not familiar with "set verbose=True". Do you mean run df.info(verbose=True)? I ran this command and checked that no gene was repeated by set(). See pic. below. 
![91697889779_ pic](https://github.com/zqfang/GSEApy/assets/61700436/45573538-f10b-4c8d-85d2-a0473e320196)
I need to talk with my collaborators before sharing data, or I may make a sub-dataset and upload it later.

**Reply by Cher-HAN** [reply]
Hi zq, I realized I (maybe you also) misunderstand the meaning/calculation of gene %.
The gene% should be (the position of the gene coresponding to enrichment score peak) /  (number of genes in the ranking list)
This formular is for gene set with positive enrichment score. 

In my example, 18 leadings are at position 191 to 384 in the ranking list. And because this GO term has a negative enrichment score, the leading genes are located after the enrichment score peak. Thus the position of gene coresponding to the enrichment score peak is counted from end to start whcih is total number 389 - position 191 + 1,  
gene % = (389 - 191 + 1)/389 = 51.16 %.
![161698084870_ pic](https://github.com/zqfang/GSEApy/assets/61700436/73e89a29-53c6-48ab-8469-b9740ebfa192)

**Reply by zqfang** [reply]
Oooo.....h Yes. You'er right. The answer is updated now!
